src/super_gradients/training/models/model_factory.py

```diff
@@ -0,0 +1,102 @@
+from typing import Union
+
+import torch
+import torch.nn as nn
+
+from super_gradients.training import utils as core_utils
+from super_gradients.training.models import SgModule
+from super_gradients.training.models.all_architectures import ARCHITECTURES
+from super_gradients.training.pretrained_models import PRETRAINED_NUM_CLASSES
+from super_gradients.training.utils.checkpoint_utils import load_checkpoint_to_model, load_pretrained_weights, \
+    read_ckpt_state_dict
+
+
+class SgNetsFactory:
+    @classmethod
+    def get(cls, architecture: Union[str, nn.Module], arch_params={}, checkpoint_params={}, *args,
+            **kwargs) -> nn.Module:
+        """
+        :param architecture:               Defines the network's architecture from models/ALL_ARCHITECTURES
+        :param arch_params:                Architecture H.P. e.g.: block, num_blocks, num_classes, etc.
+        :param checkpoint_params:          Dictionary like object with the following key:values:
+
+            strict_load:                See StrictLoad class documentation for details.
+            load_backbone:              loads the provided checkpoint to net.backbone instead of net
+            checkpoint_path:   The path to the external checkpoint to be loaded. Can be absolute or relative
+                                               (ie: path/to/checkpoint.pth). If provided, will automatically attempt to
+                                               load the checkpoint even if the load_checkpoint flag is not provided.
```

Comment:
can we have another flag "pretrained". if true it will load the model from the default path defined for this model (unless overridden by checkpoint_path.
so that if someone just wants a pretrained yolox small, they cat just set the flag true